lib/project_types/script/layers/application/create_script.rb

```diff
@@ -65,15 +63,17 @@ def install_dependencies(ctx, language, script_name, project_creator)
           def in_new_directory_context(ctx, directory)
             initial_directory = ctx.root
             begin
-              ctx.mkdir_p(directory)
-              ctx.chdir(directory)
+              Infrastructure::ScriptProjectRepository.create_project_directory(ctx: ctx, directory: directory)
               yield
             rescue
-              ctx.chdir(initial_directory)
-              ctx.rm_r(directory)
+              Infrastructure::ScriptProjectRepository.delete_project_directory(
```

Comment:
Great catch! In order to remedy this, I've added an extra rescue statement so that we dont delete the directory if the error is "ScriptProjectAlreadyExistsError"